zipline/finance/trading.py

```diff
@@ -504,53 +507,49 @@ def __init__(self, period_start, period_end,
         # copied to algorithm's environment for runtime access
         self.arena = 'backtest'
 
-        if env is not None:
-            self.update_internal_from_env(env=env)
+        if cal is not None:
+            self.update_internal_from_cal(cal=cal)
 
-    def update_internal_from_env(self, env):
+    def update_internal_from_cal(self, cal):
 
         assert self.period_start <= self.period_end, \
             "Period start falls after period end."
 
-        assert self.period_start <= env.last_trading_day, \
+        assert self.period_start <= cal.last_trading_day, \
             "Period start falls after the last known trading day."
-        assert self.period_end >= env.first_trading_day, \
+        assert self.period_end >= cal.first_trading_day, \
             "Period end falls before the first known trading day."
 
-        self.first_open = self._calculate_first_open(env)
-        self.last_close = self._calculate_last_close(env)
-
-        start_index = env.get_index(self.first_open)
-        end_index = env.get_index(self.last_close)
+        self.first_open = self._calculate_first_open(cal)
+        self.last_close = self._calculate_last_close(cal)
```

Comment:
Note to self: these fields seem more like they should be SimulationParameters, not attributes of the environment